src/Microsoft.AspNet.Mvc.Core/Internal/ResponseContentTypeHelper.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.Mvc.Core;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNet.Mvc.Internal
+{
+    public static class ResponseContentTypeHelper
+    {
+        public static MediaTypeHeaderValue GetContentType(
+            MediaTypeHeaderValue actionResultContentType,
+            string httpResponseContentType,
+            MediaTypeHeaderValue defaultContentType)
+        {
+            if (defaultContentType == null)
+            {
+                throw new ArgumentNullException(nameof(defaultContentType));
+            }
+
+            if (defaultContentType.Encoding == null)
+            {
+                throw new InvalidOperationException(
+                    Resources.FormatDefaultContentTypeMustHaveEncoding(defaultContentType.ToString()));
```

Comment:
This should be an `ArgumentException` instead. Look at where `Resources.ModelBinderUtil_ModelMetadataCannotBeNull` is used in `ModelBindingHelper.ValidateBindingContext()`.